offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/projects/ProjectMainConnector.groovy

```diff
@@ -46,16 +48,23 @@ class ProjectMainConnector implements CreateProjectDataSource, CreateProjectSpac
   private final OpenBisClient openbisClient
 
   /**
+
    * Constructor for a ProjectMainConnector
    * @param projectDbConnector a connector enabling interaction with the project database
    * @param openbisClient an openBIS client API object
-   * 
-   * @see Connection
    */
   ProjectMainConnector(ProjectDbConnector projectDbConnector, OpenBisClient openbisClient) {
     this.projectDbConnector = projectDbConnector
     this.openbisClient = openbisClient
   }
+  
+  public List<ProjectSpace> listSpaces() {
+    List<ProjectSpace> res = new ArrayList<>()
+    for(String spaceName : openbisClient.listSpaces()) {
+      res.add(new ProjectSpace(spaceName))
+    }
+    return res
+  }
 
```

Comment:
We only want that once at the construction of the service. Otherwise, we will create an unnecessary load on openbis.

And: Java Docs are missing :)

Reply:
Please return a new list reference as well

Reply:
like this?